### PR TITLE
added LastValEqual component class

### PR DESCRIPTION
**Pull request by pluflou**
Added new LastValEqual class to gfosd.components.equality_constraints.
Added LastValEqual import to components/__init__.py

### CODE REVIEW

File: gfosd/components/equality_constraints.py [hunk 1]
```diff
@@ -27,7 +27,33 @@ def _make_B(self):
...
+        super().__init__(*args, **kwargs)
+        # always retain helper variable
+        self._has_helpers = True
+
+    def _make_A(self):
+        super()._make_A()
+        super()._make_B()
+        super()._make_c()
+        self._A = sp.bmat([
+           [self._A.tocsr()[self._A.shape[1]-1]],
```
Comment (bmeyers) [inline gfosd/components/equality_constraints.py]: Let's please change to `self._A.tocsr()[-1]` making of standard Pythonic indexing methods.

File: gfosd/components/equality_constraints.py [hunk 2]
```diff
@@ -27,7 +27,33 @@ def _make_B(self):
...
+            [sp.dok_matrix((1, self._A.shape[1]))]
+        ])
 
+    def _make_B(self):
+        self._B = sp.bmat([
+           [self._B.tocsr()[self._B.shape[1]-1]],
```
Comment (bmeyers) [inline gfosd/components/equality_constraints.py]: same request here

File: gfosd/components/equality_constraints.py (hunk 1 above)
Comment (bmeyers) [reply gfosd/components/equality_constraints.py]: what you wrote will work, to clear clear, but this change will make the code a bit more readable

File: gfosd/components/equality_constraints.py [hunk 3]
```diff
@@ -27,7 +27,33 @@ def _make_B(self):
...
+            [sp.coo_matrix(([1], ([0], [self._B.shape[1]-1])), shape=(1, self._B.shape[1]))]
```
Comment (bmeyers) [inline gfosd/components/equality_constraints.py:50]: you are correct that negative indexing does _not_ work in this instance, so this is correct 👍

**Review by bmeyers — CHANGES_REQUESTED** [review]

File: gfosd/components/equality_constraints.py [hunk 4]
```diff
@@ -27,7 +27,33 @@ def _make_B(self):
...
+        ])
+
+    def _make_c(self):
+        self._c = np.concatenate([np.atleast_1d(self._c[0]),
```
Comment (bmeyers) [inline gfosd/components/equality_constraints.py]: The equality constraint here should be on the last entries of `x` and `z`, so this should in fact be `self._c[-1]`. The value is the same, `self._c[0] == self._c[-1]`, but we should fix the index for the sake of clarity, I think.

**Comment by pluflou** [comment]
Thanks @bmeyers. I made the pushed the requested changes.

**Review by bmeyers — APPROVED** [review]

**Pull request merged**